plugin/discovery/meta.go

```diff
@@ -0,0 +1,28 @@
+package discovery
+
+import (
+	"github.com/blang/semver"
```

Comment:
I didn't know hashicorp had its own semver library, so I suppose the primary "benefit" is that this one shows up earlier in Google search results for "golang semver" :grinning:

I don't have a strong opinion, honestly. The two libraries look pretty equivalent on the surface for our purposes, though this third-party one is well-documented, [well-used](https://godoc.org/github.com/blang/semver?importers), and has a compelling list of "Why should I use this" arguments. I don't know how many of those same arguments apply to the home-grown one.

I'm happy to switch it to the hashicorp-maintained one if you think that's important.